### DISCUSSION TITLE
WebApp filtrowanie IP

### DISCUSSION THREAD

**Discussion by adriantheking** [AzureCommunityPL/group#343] [Pomoc] [answered]
Cześć,

Potrzebuję od Was informacji czy jest możliwość(jeżeli jest, to mile widziane hasła jak to zrobić) zabezpieczenia poprzez filtrowanie IP WebApp ale tylko na jednym wybranym endpoincie ? Wszystko co dotychczas znalazłem dotyczy filtrowania IP na cała aplikację.

**Comment by ptrstpp950** [comment, marked as answer]
Nie da się, chyba że zrobisz to w kodzie aplikacji a nie w ustawieniach Web App.
Natomiast nie istnieje żaden problem, aby:
- mieć jeden App Plan (za to głównie płacisz)
- dostawić 1 lub 2 reverse proxy (np Nginx) dla Twojej apki zgodnie z potrzebami i wygodą na tym samym App Plan.
Jeżeli używasz WebApp for containers to możesz postawić 2 kontenery wewnątrz jednej web appki z czego jedna to reverse proxy (np: Nginx)

**Reply by ptrstpp950** [reply]
A teraz odpowiedź z gwiazdką i hackowaniem, czyli nie rób tego w domu jak już zupełnie nie musisz :)
Jeżeli używasz Web App to w teorii ciągle po spodem jest IIS:
- w applicationHost.config możesz ustawić ipFilter per endpoint ([przykład](https://dejanstojanovic.net/powershell/2018/march/setting-ip-address-and-domain-filtering-in-iis-using-powershell-script/))
- możesz zrobić deployment zmieniający applicationHost.config z użyciem [instrukcji](https://github.com/uglide/azure-content/blob/master/articles/app-service-web/web-sites-transform-extend.md#:~:text=Azure%20App%20Service%20web%20app%20advanced%20config%20and,extensions%20to%20enable%20custom%20web%20app%20administration%20actions.)

Podkreślam to jest hacking i nie daję za to gwarancji oraz 100% że ciągle działa i że jutro będzie działało 🙊🙉🐵

**Reply by mgrabarz** [reply]
Ja przekornie zaproponuję rozwiązanie prostsze niż stawianie nginxów, używanie docker compose albo dłubania w IIS (który jest tylko w AppService Windows kind). Kto to potem utrzyma, jak będziesz takie rozwiązania na duct tape tworzył? :)

Postaw przed AppService Azure Front Doora. Dla poszczególnych host name / ścieżek stawiasz osobne regułki (WAF, przekierowania, filtrowanie po IP). Na AppService robisz filtrowanie, że tylko Front Door może tam wchodzić (Sevice Tag + FDID).

**Reply by adriantheking** [reply]
@mgrabarz Dzięki ! Przyjrzę się temu bardziej, z tego co napisałeś to jest  to czego szukałem

**Comment by piotrgwiazda** [comment]
+ dla Front Door

Rzecz w tym, że AppService jest odpowiednio droższy niż np. Kubernetes z
podobnym CPU i RAM bo dodatkowo obsługuje warstwę sieciową, DNS, SSL itp.
dla "apki".
Teraz dopłacamy za Front Door i tam pójdą certy SSL (trzeba rozszyfrować
ruch zeby go filtrować w 7 warstwie) pojawi się tam jakaś kwestia DNS itp.

Oczywiście to jest OK ale pojawia się płacenie dwa razy za to samo.
Może ta "apka" to już nie jest "prosta apka" i potrzeba innych narzędzi.


wt., 21 wrz 2021, 14:29 użytkownik adriantheking ***@***.***>
napisał:
